src/Pages/Layout.tsx

```diff
@@ -24,10 +26,20 @@ export default function Layout() {
     const readNotifications = useSelector<RootState, number>(s => s.login.readNotifications);
     const dms = useSelector<RootState, RawEvent[]>(s => s.login.dms);
     const prefs = useSelector<RootState, UserPreferences>(s => s.login.preferences);
+    const pub = useEventPublisher();
     useLoginFeed();
 
     useEffect(() => {
         if (relays) {
+            const nip42AuthEvent = async (event: NIP42AuthChallenge) => {
```

Comment:
I think it would be simpler to just attach a function to `System` at startup which can handle these auth methods, instead of using the window event listener

Reply:
Yeah, that does seem cleaner.

I just made some changes and pushed.